### PR TITLE
Show available stores on snap registration page

### PR DESCRIPTION
**Pull request by Lukewh**
Fixes https://github.com/canonicalltd/snap-squad/issues/840
Fixes https://github.com/canonicalltd/snap-squad/issues/841
Fixes https://github.com/canonicalltd/snap-squad/issues/842

# QA
- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/register-snap
- If you don't have access to any stores, you should see the original page
- If you do have stores, you should see a dropdown that contains the available store, including the Global store

### No stores
screenshot_2019-01-14 snapcraft - snaps are universal linux packages 1

### Stores
screenshot_2019-01-14 snapcraft - snaps are universal linux packages

### CODE REVIEW

File: webapp/publisher/snaps/views.py [hunk 1]
```diff
@@ -677,6 +690,18 @@ def redirect_post_register_name():
...
+        # We don't use the data from this endpoint.
+        # It is mostly used to make sure the user has signed
+        # the terms and conditions.
+        user = api.get_account(flask.session)
+    except ApiResponseErrorList as api_response_error_list:
+        return _handle_error_list(api_response_error_list.errors)
+    except ApiError as api_error:
+        return _handle_errors(api_error)
+
+    available_stores = logic.filter_available_stores(user["stores"])
```
Comment (tbille) [inline webapp/publisher/snaps/views.py]: No need to run this function if the request works. Put it after line 707

File: webapp/publisher/snaps/views.py [hunk 2]
```diff
@@ -639,6 +639,18 @@ def redirect_get_register_name():
 @publisher_snaps.route("/register-snap")
 @login_required
 def get_register_name():
+    try:
+        # We don't use the data from this endpoint.
+        # It is mostly used to make sure the user has signed
+        # the terms and conditions.
```
Comment (tbille) [inline webapp/publisher/snaps/views.py]: this is not really true you use at this line: `available_stores = logic.filter_available_stores(user["stores"])`

File: webapp/publisher/snaps/views.py [hunk 3]
```diff
@@ -677,6 +690,18 @@ def redirect_post_register_name():
 @publisher_snaps.route("/register-snap", methods=["POST"])
 @login_required
 def post_register_name():
+    try:
+        # We don't use the data from this endpoint.
+        # It is mostly used to make sure the user has signed
+        # the terms and conditions.
```
Comment (tbille) [inline webapp/publisher/snaps/views.py]: Can you give me a use case that would make it fail?
This is handled at login. When the user logins he should be redirected before even getting into this function thanks to the annotation `@login_required`

File: webapp/publisher/snaps/views.py (hunk 1 above)
Comment (Lukewh) [reply webapp/publisher/snaps/views.py]: :+1:

File: webapp/publisher/snaps/views.py (hunk 2 above)
Comment (Lukewh) [reply webapp/publisher/snaps/views.py]: :+1:

File: webapp/publisher/snaps/views.py (hunk 3 above)
Comment (Lukewh) [reply webapp/publisher/snaps/views.py]: :+1:

**Comment by therealjuan** [comment]
👍 🎉 :shipit:  👍 🎉 :shipit:  LGTM 👍 🎉 :shipit:  👍 🎉 :shipit:  👍 🎉 :shipit:

File: webapp/publisher/snaps/views.py [hunk 4]
```diff
@@ -639,6 +639,10 @@ def redirect_get_register_name():
 @publisher_snaps.route("/register-snap")
 @login_required
 def get_register_name():
+    user = api.get_account(flask.session)
```
Comment (tbille) [inline webapp/publisher/snaps/views.py]: please suround with `try...expect`

File: webapp/publisher/snaps/views.py [hunk 5]
```diff
@@ -695,6 +700,10 @@ def post_register_name():
             registrant_comment=registrant_comment,
         )
     except ApiResponseErrorList as api_response_error_list:
+        user = api.get_account(flask.session)
```
Comment (tbille) [inline webapp/publisher/snaps/views.py]: please suround with `try...expect`

File: webapp/publisher/snaps/views.py (hunk 4 above)
Comment (Lukewh) [reply webapp/publisher/snaps/views.py]: I went too far the other way!

File: webapp/publisher/snaps/views.py (hunk 5 above)
Comment (Lukewh) [reply webapp/publisher/snaps/views.py]: I went too far the other way!

**Pull request merged**